java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/util/HexString.java

```diff
@@ -1,100 +1,202 @@
-/**
- *    Copyright (c) 2008 The Board of Trustees of The Leland Stanford Junior
- *    University
- *
- *    Licensed under the Apache License, Version 2.0 (the "License"); you may
- *    not use this file except in compliance with the License. You may obtain
- *    a copy of the License at
- *
- *         http://www.apache.org/licenses/LICENSE-2.0
+package org.projectfloodlight.openflow.util;
+
+import io.netty.util.internal.EmptyArrays;
+
+/** Utility method to convert hexadecimal string from/to longs and byte arrays.
  *
- *    Unless required by applicable law or agreed to in writing, software
- *    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
- *    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
- *    License for the specific language governing permissions and limitations
- *    under the License.
- **/
+ * @author Andreas Wundsam <andreas.wundsam@bigswitch.com>
+ */
+public final class HexString {
 
-package org.projectfloodlight.openflow.util;
+    private HexString() {}
 
-import org.projectfloodlight.openflow.types.U8;
+    /* ================================================================================
+     * Implementation note:
+     * These implementations are optimized for small-O efficiency and thus rather ugly.
+     *
+     * When making changes, make sure *every line* is covered by a unit test.
+     *
+     * Do not use this as a blue print for normal code.
+     * ================================================================================
+     */
+
+    private final static char[] CHARS =
+        { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };
 
-public class HexString {
     /**
      * Convert a string of bytes to a ':' separated hex string
      *
      * @param bytes
      * @return "0f:ca:fe:de:ad:be:ef"
      */
     public static String toHexString(final byte[] bytes) {
-        int i;
```

Comment:
This diff is useless since I rewrote the code - better look at the new version of the file.